resilience4j-timelimiter/src/main/java/io/github/resilience4j/timelimiter/internal/TimeLimiterImpl.java

```diff
@@ -137,4 +140,38 @@ private void publishEvent(TimeLimiterEvent event) {
             LOG.warn("Failed to handle event {}", event.getEventType(), e);
         }
     }
+
+    /**
+     * Completes CompletableFuture with {@link TimeoutException}.
+     */
+    static final class Timeout {
+
+        private static final ScheduledThreadPoolExecutor delayer;
```

Comment:
Shouldn't be  non-static ScheduledThreadPoolExecutor taken from TimeLimiterConfig? Optional parameter with default value like this one here.

Reply:
`delayer` just completes the future exceptionally without doing anything. So I think it is ok with one thread. 
But, I found warnings about `CompletableFuture` in @Romeh 's [link](https://crondev.blog/2017/01/23/timeouts-with-java-8-completablefuture-youre-probably-doing-it-wrong/). Subsequent executions can be run on `delayer` if timeout occurs. 
In jdk9 implementation, it doesn't care about this and leave it to the users. And in our library, users can compose `TimeLimiter` and `ThreadPoolBulkhead` as @RobWin said in slack. so I think we don't have to worry about it either. What do you think?

Reply:
Maybe some test could show that it can work parallel without loss, not like a "synchronized" one at once?

Reply:
We could solve it like: `Retry.decorateCompletionStage`.
The method has a `ScheduledExecutorService scheduler` parameter which allows the user (per call) to configure a scheduler which performs the async retry.

Reply:
Changed to getting `ScheduledExecutorService` from parameter like `Retry.decorateCompletionStage`.